### PR TITLE
Adds Video Support

### PR DESCRIPTION
**Pull request by stuartmscott**
This feature is basically a duplicate of image, enabling videos to
be included in an epub.

Sample mp4 from https://filesamples.com/samples/video/mp4/sample_640x360.mp4

### CODE REVIEW

File: fetchmedia.go [hunk 1]
```diff
@@ -74,22 +74,25 @@ func (g grabber) fetchMedia(mediaSource, mediaFolderPath, mediaFilename string)
...
 		// There shouldn't be any problem with the writer, but the reader
 		// might have an issue
 		return "", &FileRetrievalError{Source: mediaSource, Err: err}
 	}
+
 	// Detect the mediaType
-	mime, err := mimetype.DetectReader(&buffer)
+	r, err := filesystem.Open(mediaFilePath)
+	if err != nil {
+		return "", err
```
Comment (stuartmscott) [inline fetchmedia.go:88]: I'm not sure if this should be `return "", err`, `return "", &FileRetrievalError{Source: mediaSource, Err: err}`, or `panic(err)` - please advise

**Comment by stuartmscott** [comment]
The `TestEpubValidity/MemoryFS` failure is due to the in-memory filesystem using `bytes.Buffer` to hold the file contents. When the mime detector peeks at the data it increases the offset inside the buffer, and therefore subsequent reads are not reading the whole file, leading to "Image file header may be corrupted." and " Corrupted image file encountered." EPubCheck errors.

**Comment by stuartmscott** [comment]
Why does `os.Remove` not work on Windows?

**Comment by bmaupin** [comment]
@stuartmscott Thanks for your contribution! @owulveryck Feel free to take a look at this if you'd like. Otherwise I'll try to take a look at it next time I get a chance, probably some time after the new year.

**Comment by owulveryck** [comment]
I will have a closer look, but I guess that it won't be before next year.
So far, Iike the change of the memory filesystem, to get rid of the bytes.buffer that was used as a commodity; I guess that this PR should bring better performances.

File: fetchmedia.go (hunk 1 above)
Comment (owulveryck) [reply fetchmedia.go:88]: I guess that panic should be raised.
But I wonder why you change this method?
What’s wrong with the multiwriter?

**Review by owulveryck — COMMENTED** [review]
LGTM, I am waiting the reply to the comment I made to the change of the fetchMedia function.

File: fetchmedia.go (hunk 1 above)
Comment (stuartmscott) [reply fetchmedia.go:88]: The multi-writer wrote to `bytes.Buffer` which is held in-memory and can cause out-of-memory crashes with large files such as videos. This switches to a temporary file which should be capable of handling large files.

**Review by owulveryck — APPROVED** [review]

**Comment by owulveryck** [comment]
I will tag this tomorrow

**Comment by owulveryck** [comment]
Thank you very much for this contribution.

**Comment by stuartmscott** [comment]
You're very welcome, glad to see it land!

**Comment by bmaupin** [comment]
Thanks to both of you!

**Comment by stuartmscott** [comment]
Here's the result if you're interested; [read online](https://convey.earth/digest?edition=2021-12) or [download](https://convey.earth/digest/Convey-Digest-2021-12.epub)

**Pull request merged**